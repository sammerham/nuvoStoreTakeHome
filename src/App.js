import { React, useState, useEffect } from 'react';
import StoreContext from './storeContext';
import axios from 'axios';
import Intodcudion from './components/Intodcudion';
import ItemList from './components/ItemList';
import ItemDetails from './components/ItemDetails';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
const URL = 'https://fakestoreapi.com/products?limit=5'


/*
 * App
 *
 * This is the page where you rednder app.
 * State: 
 *      Prooducts: [{}, {}, ....]
 *      CurrItem : {}
 *      currItemClicked Boolean
 *      clickedId :id
 * props:
        None
  
 * App 
 */


function App() {

  const [products, setProducts] = useState([]);
  const [currItem, setCurrItem] = useState(null);
  const [currItemClicked, setCurrItemClicked] = useState(false);
  const [clickedId, setClickedId] = useState(null);


  //side effect to get products from api on load;
  useEffect(() => {
    async function fetchProducts() {
      const products = await axios.get(URL);
      setProducts(data => products.data);
    };
    fetchProducts();
  }, []);



  return (
    <StoreContext.Provider value=
      {{
        products,
        currItem,
        currItemClicked,
        setCurrItem,
        setCurrItemClicked,
        clickedId,
        setClickedId
      }}
    >
      <div style={{ backgroundColor: 'lightgray' }}>
        <Container >
          <Row>
            <Col>
              <Intodcudion />
            </Col>
          </Row>
          <Row>
            <Col sm md={6}>
              <ItemList />
            </Col>
            <Col sm md={6}>
              {clickedId && clickedId === currItem.id
                ?
                <ItemDetails />
                :
                null
              }
            </Col>
          </Row>
        </Container>
      </div>
    </ StoreContext.Provider >
  );
}

export default App;
