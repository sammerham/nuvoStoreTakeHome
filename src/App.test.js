import React from 'react'
import { screen, render, fireEvent } from '@testing-library/react';


import App from './App';


// smoke test
test('renders without crashing', () => {
  render(<App />);
});



// snapShot
test("matches snapshot", function () {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});


test("checking if title and intro show on main app", function () {
  render(<App />);
  expect(screen.getByText('Nuvo Store')).toBeInTheDocument()
  expect(screen.getByText('perspiciatis quisquam', { exact: false })).toBeVisible()
});

test("checking details box show when clicking product title", async function () {
  render(<App />);
  const title = await screen.findByText('Fjallraven', { exact: false })
  fireEvent.click(title)
  await screen.findByText('$', { exact: false })
  expect(screen.getByText('$', { exact: false })).toBeInTheDocument()
  const btn = await screen.findByText('Add to Cart', { exact: false })
  fireEvent.click(btn)
  expect(screen.queryByText('$', { exact: false })).not.toBeInTheDocument()
});

