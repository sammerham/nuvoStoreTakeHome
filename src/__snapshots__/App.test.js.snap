// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    style="background-color: lightgray;"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                <h1
                  class="sc-bdvvtL dyRZpS"
                >
                  Nuvo Store
                </h1>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col"
              >
                <p
                  class="sc-gsDKAQ dMhkiz"
                >
                  Lorem ipsum dolor sit amet consectetur adipisicing elit.Laborum ipsum laudantium possimus dolore atque perspiciatis quisquam,quasi nemo quis ipsam sunt dolorem molestiaeporro officiis.Deleniti ipsa tenetur aspernatur sunt?
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-6 col-sm"
        >
          <div />
        </div>
        <div
          class="col-md-6 col-sm"
        />
      </div>
    </div>
  </div>
</div>
`;
